askup/views.py

```diff
@@ -257,6 +258,39 @@ def user_profile_view(request, user_id):
             'own_last_week_incorrect_answers': get_student_last_week_incorrect_answers_count(
                 profile_user.id
             ),
+            'user_organizations': get_user_organizations_string(profile_user),
+            'rank_list': rank_list
+        },
+    )
+
+
+@login_required
+def user_profile_rank_list_view(request, user_id):
+    """Provide the user profile rank list view."""
+    profile_user = get_object_or_404(User, pk=user_id)
+    rank_list = get_user_profile_rank_list(profile_user.id, request.user.id)
+    return render(
+        request,
+        'askup/user_profile.html',
+        {
+            'profile_user': profile_user,
```

Comment:
Please use `user_id ` instead of `profile_user.id` each time.

Reply:
done